.ci/smoke-tests.groovy

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env groovy
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-aws-lambda'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    AWS_ACCOUNT_SECRET = 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+    EC_KEY_SECRET = 'secret/observability-team/ci/elastic-cloud/observability-pro'
+    TERRAFORM_VERSION = '1.2.3'
+    CREATED_DATE = "${new Date().getTime()}"
+    SLACK_CHANNEL = "#apm-serverless"
+    SMOKETEST_VERSIONS = "${params.SMOKETEST_VERSIONS}"
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '100', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+  }
+  parameters {
+    string(name: 'SMOKETEST_VERSIONS', defaultValue: 'latest', description: 'Run smoke tests using following APM versions')
+  }
+  stages {
+    stage('Checkout') {
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", shallow: false)
+        setEnvVar('GO_VERSION', readFile("${BASE_DIR}/.go-version").trim())
```

Comment:
This was actually copied over from smoke test jobs in other repository. Is there any value in using it in this repository ?

I think we're always targeting the latest version of Go and the user should not need to compile the extension.

Reply:
it's used by the `withGoEnv` step implicitly, although the same logic happens internally, `withGoEnv` calls https://github.com/elastic/apm-pipeline-library/blob/4bf7553413212e90a142c649ebb2ebac7b4a5503/vars/goDefaultVersion.groovy#L25-L33

you can remove it if you prefer, though it works as is and it's harmless.

Reply:
Removed :+1: